obp/ope/estimators.py

```diff
@@ -537,6 +540,68 @@ def estimate_interval(
             random_state=random_state,
         )
 
+    def _estimate_mse_upper_bound(
+        self,
+        reward: np.ndarray,
+        action: np.ndarray,
+        pscore: np.ndarray,
+        action_dist: np.ndarray,
+        position: Optional[np.ndarray] = None,
+        max_reward_value: Optional[Union[int, float]] = None,
+    ) -> float:
+        """Estimate the upper bound of the mean-squared-error of IPW with a given clipping hyperparameter.
+
+        Parameters
+        ----------
+        reward: array-like, shape (n_rounds,)
+            Reward observed in each round of the logged bandit feedback, i.e., :math:`r_t`.
+
+        action: array-like, shape (n_rounds,)
+            Action sampled by a behavior policy in each round of the logged bandit feedback, i.e., :math:`a_t`.
+
+        pscore: array-like, shape (n_rounds,)
+            Action choice probabilities by a behavior policy (propensity scores), i.e., :math:`\\pi_b(a_t|x_t)`.
+
+        action_dist: array-like, shape (n_rounds, n_actions, len_list)
+            Action choice probabilities by the evaluation policy (can be deterministic), i.e., :math:`\\pi_e(a_t|x_t)`.
+
+        position: array-like, shape (n_rounds,), default=None
+            Positions of each round in the given logged bandit feedback.
+
+        Returns
+        ----------
+        estimated_mse_upper_bound: float
+            Estimated upper bound of MSE with a given clipping hyperparameter `lambda_`.
+            This is estimated using the automatic hyperparameter tuning procedure
+            based on Section 5 of Su et al.(2020).
+
+        """
+        n_rounds = reward.shape[0]
+        # estimate the variance of IPW with clipping
+        var_hat = np.var(
+            self._estimate_round_rewards(
+                reward=reward,
+                action=action,
+                pscore=pscore,
+                action_dist=action_dist,
+                position=position,
+            )
+        )
+        var_hat /= n_rounds
```

Comment:
(Question)
Is there a reason why the confidence interval estimation uses bootstrap and this variance estimation does not?

Reply:
No specific reason. I think we should add a way to use the sample variance to construct CI to the `estimate_interval` side later.